src/components/organisms/Navbar/Navbar.js

```diff
@@ -74,6 +75,25 @@ export default function Navbar() {
     }
 }, [router.pathname]);
 
+const [windowWidth, setWindowWidth] = useState(typeof window !== 'undefined' ? window.innerWidth : 0);
+
+useEffect(() => {
+    if (typeof window !== 'undefined') {
+        const handleResize = debounce(() => {
+            if (Math.abs(window.innerWidth - windowWidth) > 50) { // Adjust the threshold as needed
+                setNavbar(false);
+                setWindowWidth(window.innerWidth);
+            }
+        }, 300); // Adjust the debounce delay as needed
+
+        window.addEventListener('resize', handleResize);
+
+        return () => {
+            window.removeEventListener('resize', handleResize);
+        };
+    }
+}, [windowWidth]);
```

Comment:
1. I agree that the primary parameter to determine whether the open mobile navbar should close is `innerWidth`. However, I would think that the condition to determine when the open mobile navbar closes is when `innerWidth >= 976`, and not when the width difference > 50?..This causes the mobile navbar to close whenever the width adjusts by at least `50px`, which I think is not the desired effect we want.

2. I don't think `debounce()` is needed in this case because changing the screen width while the mobile navbar is open is not a very common user activity I would assume. And a delayed action of closing the navbar after 300ms is noticeable (compared to no delay) and may affect user experience.

3. I think creating a new `windowWidth` state and applying it in the dependency array is not necessary. We don't really need to keep track of a `windowWidth` if the `innerWidth` property already provides the screen width for us. Instead, all we need to do is attach the `resize` event to `window` when the Navbar component mounts, remove the `resize` event when the Navbar unmounts, and that's it. Whenever the screen width changes, the callback will be invoked.

4. I think something simple like this may do the trick?

```
const SOME_VARIABLE = 976;

useEffect(() => {
    if (typeof window !== 'undefined') {
        const handleResize = () => {
            if (window.innerWidth >= SOME_VARIABLE) {
                setNavbar(false);
            }
        };

        window.addEventListener('resize', handleResize);

        return () => {
            window.removeEventListener('resize', handleResize);
        };
    }
}, []);
```

Thoughts?